src/cbc_sdk/workload/compliance_assessment.py

```diff
@@ -0,0 +1,465 @@
+            >>> cb = CBCloudAPI(profile="example_profile")
+            >>> benchmark_sets = cb.select(ComplianceBenchmark)
+            >>> # To return all rules within a benchmark set, leave search_rules empty.
+            >>> rules = list(benchmark_sets[0].search_rules())
+            >>> print(*rules)
+            >>> # To return a single rule document, add the rule ID.
+            >>> rule = list(benchmark_sets.search_rules('00869D86-6E61-4D7D-A0A3-6F5CDE2E5753'))
+            >>> print(rule)
+        """
+        if rule_id is not None:
+            self._rule_id = rule_id
+            url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/rules/{rule_id}"
+            result = self._cb.get_object(url)
+            yield ComplianceBenchmark(self._cb, initial_data=result)
+            return
+
+        url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/rules/_search"
+        current = 0
+        max_rows = 80000
+        while True:
+            # request = self._build_request(current, max_rows)
+            resp = self._cb.post_object(url, body={})  # FIXME fix the body
+            result = resp.json()
+
+            self._total_results = result["num_found"]
+            self._count_valid = True
+
+            results = result.get("results", [])
+            for item in results:
+                yield ComplianceBenchmark(self._cb, initial_data=item)
+
+            current += len(results)
+            if max_rows > 0 and current >= max_rows:
+                break
+
+            if current >= self._total_results:
+                break
+
+    def get_set_sections(self):
+        """
+        Get Sections of the Benchmark Set.
+
+        Returns:
+            generator of ComplianceBenchmark: A generator yielding Compliance instances
+                                           for each section in the benchmark set.
+
+        Example:
+            >>> cb = CBCloudAPI(profile="example_profile")
+            >>> benchmark = cb.select(ComplianceBenchmark).set_benchmark_set_id('benchmark123')
+            >>> for section in benchmark.get_benchmark_set_sections():
+            ...     print(section.section_name, section.section_id)
+        """
+        url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/sections"
+        results = self._cb.get_object(url)
+        for item in results:
+            yield ComplianceBenchmark(self._cb, initial_data=item)
+
+    def execute_action(self, action, device_ids=None):
+        """
+        Execute a specified action on devices within on a Benchmark Set, or specific devives
+        within a Benchmark Set only.
+
+        Args:
+            action (str): The action to be executed. Available:
+                'ENABLE': Enable the object.
+                'DISABLE': Disable the object.
+                'REASSESS': Reassess the object.
+
+            device_ids (str or list, optional): IDs of devices on which the action will be executed.
```

Comment:
Don't the ids need sent as integers not strings

Reply:
Yes, but I convert them to integers anyway, rather than throw an error.

Reply:
The problem is we would source them as ints already then expect to convert them to strings then back again. I'd recommend using int still instead of str